networking/ipsec/ipsec_basic/ipsec_basic_netns/ipsec-parameter-setting.sh

```diff
@@ -0,0 +1,169 @@
+		echo "a x      Authenticated algorithm(auth):"
+		echo "             sha1,sha256,sha384,sha512,rmd160"
+		echo "t x      Authenticated algorithm(auth-trunc):"
+		echo "             sha1,sha256,sha384,sha512,rmd160"
+		echo "e x      Encryption algorithm:"
+		echo "             des,des3_ede,cast5,blowfish,aes,twofish,camellia,serpent"
+		echo "A x      Authentication encryption with associated data algorithm(AEAD):"
+		echo "             rfc4106_128,rfc4106_192,rfc4106_256"
+		echo "             rfc4309_128,rfc4309_192,rfc4309_256"
+		echo "             rfc4543_128,rfc4543_192,rfc4543_256"
+		echo "c x      Compression algorithm"
+		echo "6        run over IPv6"
+		exit 0
+	;;
+	m)	IPSEC_MODE=$OPTARG ;;
+	p)	IPSEC_PROTO=$OPTARG ;;
+	s)	IPSEC_SIZE_ARRAY="$OPTARG" ;;
+	S)	SPI=$OPTARG ;;
+	k)	VTI_KEY=$OPTARG ;;
+	a)	AALGO=$OPTARG ;;
+	t)	ATALGO=$OPTARG ;;
+	e)	EALGO=$OPTARG ;;
+	A)	AEALGO=$OPTARG ;;
+	c)	CALGO=$OPTARG ;;
+	6)	TEST_VER=6 ;;
+	*)	echo "Error: unknown option: $opt" | tee $IPSEC_PARA_LOG; exit 1 ;;
+	esac
+done
+
+# Authenticated encryption with associated data
+AEALGO=${AEALGO:-"rfc4106_128"}
+# Encryption algorithm
+EALGO=${EALGO:-""}
+# Authentication algorithm(auth)
+AALGO=${AALGO:-""}
+# Authentication algorithm(auth-trunc)
+ATALGO=${ATALGO:-""}
+# Compression algorithm
+CALGO=${CALGO:-"deflate"}
+
+IPSEC_PROTO=${IPSEC_PROTO:-"esp_aead"}
+IPSEC_MODE=${IPSEC_MODE:-"transport"}
+SPI=${SPI:-1000}
+VTI_KEY=${VTI_KEY:-10}
+IPSEC_SIZE_ARRAY="${IPSEC_SIZE_ARRAY:-10 1000 10000 65000}"
+TEST_VER=${TEST_VER:-4}
+
+get_key()
+{
+	local bits=$1
+	local bytes=$(( $bits / 8 ))
+	key=$(echo $(for i in $(seq $bytes); do echo -n '0f'; done))
+	echo 0x$key
+}
+
+case $AEALGO in
+rfc4106_128|rfc4543_128) AEALGO_KEY=$(get_key 160) ;;
+rfc4106_192|rfc4543_192) AEALGO_KEY=$(get_key 224) ;;
+rfc4106_256|rfc4543_256) AEALGO_KEY=$(get_key 288) ;;
+rfc4309_128) AEALGO_KEY=$(get_key 152) ;;
+rfc4309_192) AEALGO_KEY=$(get_key 216) ;;
+rfc4309_256) AEALGO_KEY=$(get_key 280) ;;
+esac
+
+case $EALGO in
+des) EALGO_KEY=$(get_key 64) ;;
+des3_ede) EALGO_KEY=$(get_key 192) ;;
+cast5) EALGO_KEY=$(get_key 128) ;;
+blowfish) EALGO_KEY=$(get_key 448) ;;
+aes|twofish|camellia|serpent) EALGO_KEY=$(get_key 256) ;;
+esac
+
+case $AALGO in
+sha1|rmd160) AALGO_KEY=$(get_key 160) ;;
+sha256) AALGO_KEY=$(get_key 256) ;;
+sha384) AALGO_KEY=$(get_key 384) ;;
+sha512) AALGO_KEY=$(get_key 512) ;;
+esac
+
+case $ATALGO in
+sha1|rmd160) ATALGO_KEY=$(get_key 160) ;;
+sha256) ATALGO_KEY=$(get_key 256) ;;
+sha384) ATALGO_KEY=$(get_key 384) ;;
+sha512) ATALGO_KEY=$(get_key 512) ;;
+esac
+
+# PROTO and ALG setting
+PROTO="proto $IPSEC_PROTO"
+[ "$IPSEC_PROTO" = "esp_aead" ] && PROTO="proto esp"
+
+case $IPSEC_PROTO in
+ah)
+	if [ ! -z $AALGO ];then
+		ALG='auth hmac\('$AALGO'\) '$AALGO_KEY
+	elif [ ! -z $ATALGO ];then
+		ALG='auth-trunc hmac\('$ATALGO'\) '$ATALGO_KEY" 96"
+	else
+		echo "Error: ah protocol doesn't set authentication" | tee $IPSEC_PARA_LOG
+		exit 1
```

Comment:
If I get correctly how this file is used, it's to set up the passed parameters. Which means if `exit 1` it's a test setup problem and not a kernel problem. Should it abort instead, or am I misinterpreting bash?

Reply:
Good question, the parameters are setup during the rlPhaseStartSetup, if we abort here, we may want to abort for all cases in this file. I will need to get back to you and check with the test maintainer about this.

Reply:
As a separate config file, I think the behavior is OK. It will exit 1 and print error message if a user gives the wrong parameter. While in the test, we will give the correct parameter and it should never abort here. BTW, the subtest.yml contains some example SUB_PARAM for upstream users.

Reply:
@liuhangbin thanks for the comment! I'd really like to avoid developers complaining about failures because we made a typo in the setup that no one noticed. We require a warn/aborted status for non-kernel issues (this includes failures because the test can't be set up properly) because we can filter that out and not send it to them. Does the reasoning make sense?

If both real issues and setup issues report failure then we need to keep a human in loop to review that problem, and we'd like to avoid that as much as possible as we all have a ton on different responsibilities :)

Reply:
OK, makes sense to me. I will ask the author if she want to update it.

Thanks
Hangbin

Reply:
hi @veruu I updated relevant sections in that file to abort if param settings failed, can you review again ?